admin/admin-model.php

```diff
@@ -461,4 +466,94 @@ public function update_default_lang( $slug ) {
 		$this->clean_languages_cache();
 		flush_rewrite_rules();
 	}
+
+	/**
+	 * Adds the missing language terms for the secondary language taxonomies.
+	 *
+	 * @since 3.4
+	 *
+	 * @return void
+	 */
+	public function add_missing_secondary_language_terms() {
+		$objects = $this->translatable_objects->get_secondary_translatable_objects();
+		$updated = false;
+
+		foreach ( $this->get_languages_list() as $language ) {
+			foreach ( $objects as $object ) {
+				$term_id = $language->get_tax_prop( $object->get_tax_language(), 'term_id' );
+
+				if ( ! empty( $term_id ) ) {
+					// The term exists.
+					continue;
+				}
+
+				// Create a new term.
+				$result = $this->update_secondary_language_term(
+					array(
+						'taxonomy' => $object->get_tax_language(),
+						'slug'     => $language->slug,
+						'name'     => $language->name,
+					)
+				);
+
+				if ( ! empty( $result ) ) {
+					$updated = true;
+				}
+			}
+		}
+
+		if ( $updated ) {
+			// Clear the cache, so the new `term_id` and `term_taxonomy_id` appear in the languages list.
+			$this->clean_languages_cache();
+		}
```

Comment:
I wouldn't bother to check if seomthing's has been updated and always clean the cache. Just as we do in `add_language()` and `update_language()`.

Reply:
Done in [a610c9d](https://github.com/polylang/polylang/pull/1176/commits/a610c9da49eded79f94df96cb94bc40e4b7177d7).